### ISSUE TITLE
[Feature Request] Staging UX improvements

### ISSUE THREAD

**Issue by rlabrecque** [ForkIssues/Tracker#410]
Here's a collection of minor UX improvements around staging that I'd like to see:

1. 
Add an "Unstage All" button.

2.
Move the "Stage All" button down into the "Unstaged Files" header, next to the "Stage" button. It feels weird from a UX standpoint that it's not part of that "element".

3.
Rename "[Un]Stage" to "[Un]Stage Selected"

4.
Disable/gray out "[Un]Stage[ Selected]" when there's no files selected.

**Comment by DanPristupov** [comment]
Thank you for the proposal. Even though "I'd like to see" is not a good explanation why we should have them, here's my opinion.

1. Generally people don't use "Unstage All" so often to have a dedicated button for that. There's already Stage All button which works as a toggle button.
2. Exactly. I do want to make "Stage All" less accessible and stage each file separately, because it leads to a better commit quality.
3. Stage/Unstage is clear enough. Users got used to that UI pattern. We don't have "Copy Selected", "Bold Selected", "Cut Selected" in MS Word, right?
4. That makes sense, but I don't think it's very important right now.

It looks like you just installed Fork and started posting the issues. We spent a lot of time polishing UI making it intuitive and usable. I agree that many things are still not perfect, but I think people need to use the applications for some time before proposing global UI changes.

**Comment by rlabrecque** [comment]
> It looks like you just installed Fork and started posting the issues

That's correct! I'm on the hunt for the next tool that I will be using for hours every week for the foreseeable future. Git Fork is easily the most promising. If I'm going to be using a tool for long enough, it becomes important to be involved in some form. Obviously these issues are minor, but I just wanted to make sure I got them noted down before I get too comfortable that I don't see them as issues anymore.

1. I think our use cases are a little different, I use Unstage All (in other tools) pretty frequently when I want to change the course of the next commit. For example, if I initially had a larger pending commit, and wanted to split it into two parts because it's growing too large, it's often easier to unstage all and start cherry picking files/hunks/lines with a blank stage, than unstaging specific parts. I use it MUCH more than Stage All.
2. I can get behind that, Stage All should definitely be secondary to Stage File! My typical use case for Stage All ends up being when I have a small number of changes, roughly 2-6 files, (or more if they are binary content that I don't need to look at the diff for), I usually hit Stage All to save myself time from either staging each one manually, or having to use the keyboard to select multiple/all.
3. Fair point!
4. Indeed, It's minor.

**Comment by pavolmarkovic-serato** [comment]
Vote up for point 4.

I this this creates confusion with expectations.
When I have bunch of files and I want to stage them all, clicking Stage does nothing.
If it was disabled, it means that there's an expected condition which has not been fulfilled yet.

Simple rule is that only those controls that can actually trigger an action should be enabled.
